src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AbbreviatedNamesFunctions.java

```diff
@@ -13,24 +13,34 @@
 
     private static final String ANY_LETTER_INCLUDING_UNICODE_MATCHER = "\\p{L}\\p{M}*+";
     private static final String FULL_STOP_SPACE_MATCHER = "\\.\\s+";
+    private static final String APOSTROPHE_SPACE_MATCHER = "\'\\s+";
+
     private static final String FULL_STOP_SPACE_BETWEEN_NAMES_PATTERN = ANY_LETTER_INCLUDING_UNICODE_MATCHER + FULL_STOP_SPACE_MATCHER + ANY_LETTER_INCLUDING_UNICODE_MATCHER;
+    private static final String APOSTROPHE_SPACE_BETWEEN_NAMES_PATTERN = ANY_LETTER_INCLUDING_UNICODE_MATCHER + APOSTROPHE_SPACE_MATCHER + ANY_LETTER_INCLUDING_UNICODE_MATCHER;
+
     private static final Pattern FULL_STOP_SPACE_PATTERN_REGEX = Pattern.compile(FULL_STOP_SPACE_BETWEEN_NAMES_PATTERN);
+    private static final Pattern APOSTROPHE_SPACE_PATTERN_REGEX = Pattern.compile(APOSTROPHE_SPACE_BETWEEN_NAMES_PATTERN);
 
-    private static final String FULL_STOP_SPACE_NEGATIVE_LOOK_BEHIND = "(?<!(\\.|\\s))";
-    private static final String SPACE_NOT_PRECEDED_BY_FULL_STOP_OR_SPACE_PATTERN = FULL_STOP_SPACE_NEGATIVE_LOOK_BEHIND + "\\s+";
+    private static final String ABBREVIATION_SPACE_NEGATIVE_LOOK_BEHIND = "(?<!(\\.|'|\\s))";
+    private static final String SPACE_NOT_PRECEDED_BY_ABBREVIATION_OR_SPACE_PATTERN = ABBREVIATION_SPACE_NEGATIVE_LOOK_BEHIND + "\\s+";
 
     static boolean doesNotContainAbbreviatedNames(InputNames inputNames) {
-        if (nameContainsFullStopSpaceBetweenNames(inputNames.fullFirstName())) {
+        if (nameDoesContainAbbreviationSpaceBetweenNames(inputNames.fullFirstName())) {
             return false;
         }
-        if (nameContainsFullStopSpaceBetweenNames(inputNames.fullLastName())) {
+        if (nameDoesContainAbbreviationSpaceBetweenNames(inputNames.fullLastName())) {
             return false;
         }
-        return !nameContainsFullStopSpaceBetweenNames(inputNames.fullAliasNames());
+        return !nameDoesContainAbbreviationSpaceBetweenNames(inputNames.fullAliasNames());
     }
 
-    private static boolean nameContainsFullStopSpaceBetweenNames(String name) {
-        return FULL_STOP_SPACE_PATTERN_REGEX.matcher(name).find();
+    private static boolean nameDoesContainAbbreviationSpaceBetweenNames(String name) {
+
+        if (FULL_STOP_SPACE_PATTERN_REGEX.matcher(name).find()) {
```

Comment:
I think we can consolidate the regular expressions so that we can just use one which detects full stops or apostrophes, using something like "('|\.)".

Regular expressions are difficult so I'm happy to pair on this.

Reply:
Resolved.